cocos/core/game.ts

```diff
@@ -682,28 +685,20 @@ export class Game extends EventTarget {
         }
     }
 
-    private _stTimeWithRAF (callback) {
+    private _stTime (callback: FrameRequestCallback, through?: RequestAnimationFrame) {
+        const timeOutCallback = through ? () => through(callback) : callback;
         const currTime = performance.now();
         const elapseTime = Math.max(0, (currTime - game._lastTime));
         const timeToCall = Math.max(0, game._frameTime - elapseTime);
-        const id = window.setTimeout(() => {
-            window.requestAnimationFrame(callback);
-        }, timeToCall);
+        const id = setTimeout(timeOutCallback, timeToCall);
         game._lastTime = currTime + timeToCall;
         return id;
     }
 
-    private _stTime (callback) {
-        const currTime = performance.now();
-        const elapseTime = Math.max(0, (currTime - game._lastTime));
-        const timeToCall = Math.max(0, game._frameTime - elapseTime);
-        const id = window.setTimeout(callback, timeToCall);
-        game._lastTime = currTime + timeToCall;
-        return id;
-    }
```

Comment:
`_stTimeWithRAF` is code repetition with `_sTime`. I don't know where it got idea to "copy" the implementation... I have optimizes it anyway.  @holycanvas

Reply:
I guess some external contributors are less likely to change the existing structure of the engine to avoid introducing some other problems when modifying the engine, so they would prefer to copy the code